### PR TITLE
protocol: add missing check for txheader min/maxtime vs. block timestamp

### PR DESCRIPTION
**Pull request by bobg**
Backported from https://github.com/chain/chain/pull/1223 and https://github.com/chain/chain/pull/1224

### CODE REVIEW

File: core/txbuilder/finalize.go [hunk 1]
```diff
@@ -45,6 +45,9 @@ func FinalizeTx(ctx context.Context, c *protocol.Chain, s Submitter, tx *legacy.
 	if err != nil {
 		return errors.Wrap(err, "tx rejected")
 	}
+	if tx.Tx.MaxTimeMs > 0 && tx.Tx.MaxTimeMs < c.TimestampMS() {
```
Comment (kr) [inline core/txbuilder/finalize.go]: Is this part absolutely necessary? I'd prefer to keep change to point releases as small and conservative as possible.

File: protocol/block.go [hunk 2]
```diff
@@ -88,6 +88,17 @@ func (c *Chain) GenerateBlock(ctx context.Context, prev *legacy.Block, snapshot
 			continue
 		}
 
+		// Filter out transactions that are not yet valid, or no longer
+		// valid, per the block's timestamp.
+		if tx.Tx.MinTimeMs > 0 && tx.Tx.MinTimeMs > b.TimestampMS {
+			// TODO(bobg): log this?
```
Comment (kr) [inline protocol/block.go]: I don't think these are necessary. Same comment as in the PR on main.

File: core/txbuilder/finalize.go (hunk 1 above)
Comment (bobg) [inline core/txbuilder/finalize.go]: Ha, was just having this discussion with @jeffomatic in Slack. Your opinion tipped the balance; removed.

**Comment by kr** [comment]
LGTM

**Pull request merged**